usr/share/rear/build/GNU/Linux/390_copy_binaries_libraries.sh

```diff
@@ -20,7 +20,18 @@ function copy_binaries () {
         # Continue with the next one if a binary is empty or contains only blanks:
         contains_visible_char "$binary" || continue
         if ! cp $verbose --archive --dereference --force "$binary" "$destdir" 1>&2 ; then
-            Error "Failed to copy '$binary' to '$destdir'"
+            # When a binary should be copied where its target already exists as dangling symlink
+            # e.g. when /sbin/lvcreate should be copied to /tmp/rear.XXX/rootfs/bin/lvcreate
+            # but there is already the dangling symlink /tmp/rear.XXX/rootfs/bin/lvcreate -> lvm
+            # because its link target was not yet copied into the recovery system
+            # cf. "create LVM symlinks" in build/GNU/Linux/005_create_symlinks.sh
+            # then cp fails (regardless of the --force option) with an error message like
+            # cp: not writing through dangling symlink '/tmp/rear.XXX/rootfs/bin/lvcreate'
+            # So we skip cp errors here but add the affected binary to REQUIRED_PROGS
+            # to verify later that the binary actually exists in the recovery system
+            # regardless what the reason was why cp failed here:
+            Debug "copy_binaries skipped binary '$binary' (the binary gets verified later via REQUIRED_PROGS)"
+            REQUIRED_PROGS+=( $( basename "$binary" ) )
```

Comment:
@jsmeix I suppose that `REQUIRED_PROGS` was chosen intentionally  above `PROGS` to bail out when the executable was not found?

Reply:
@gdha 
thank you for your review!

Yes, REQUIRED_PROGS is used intentionally to error out later at
```
Testing that each program in the REQUIRED_PROGS array can be found as executable command within the recovery system
```
because for binaries in PROGS copy_binaries is only called when it exists in the original system
so adding binaries where cp failed to REQUIRED_PROGS won't add binaries
that cannot be copied because they do not exist in the original system.

I had tested things with `PROGS+=( lvcreate qqqq )` to ensure
there are no false error exits in case of non-existent binaries.

I think the root cause of the particular `PROGS+=( lvcreate )` isssue
is that "create LVM symlinks" in build/GNU/Linux/005_create_symlinks.sh
does not do things cleanly (i.e. I think it creates dangling symlinks and
I think it should better create valid symlinks i.e. symlinks with targets)
but currently I do not have free "mind capacity" to analyze how all that
is meant to work (I fear I find myself within "yet another can of worms" ;-)

So I decided to implement a simple and generic fail safe behaviour
of copy_binaries() which I tried via this pull request.

We never got issues with the hard error exit in copy_binaries()
```
if ! cp $verbose --archive --dereference --force "$binary" "$destdir" 1>&2 ; then
    Error "Failed to copy '$binary' to '$destdir'"
fi
```
so I think we can assume that `cp` basically never fails in practice
but if it fails (as with `PROGS+=( lvcreate )`) it falsely errors out
so I think DebugPrint plus REQUIRED_PROGS test is sufficient
to error out if something actually went wrong with needed binaries.